### ISSUE TITLE
Anthother Check Fail Can Be Triggered in BlockLSTM

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#58176]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.12.0-dev20221018

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.8

### CUDA/cuDNN version

CUDA 11.5

### Current Behaviour?

```shell
Another check-fail can be triggered. This seems to be different from https://github.com/tensorflow/tensorflow/issues/58175
```

### Standalone code to reproduce the issue

```shell
import os
os.environ['TF_ENABLE_ONEDNN_OPTS'] = '0'
import tensorflow as tf
import numpy as np
print(tf.__version__)
for _ in range(20):
    try:
        with tf.device("GPU:0"):
            forget_bias = -121.22699269620765
            cell_clip = -106.82307555235684
            use_peephole = False
            seq_len_max = tf.saturate_cast(tf.random.uniform([13, 11, 0], minval=0, maxval=64, dtype=tf.int64), dtype=tf.int64)
            x = tf.random.uniform([1, 3, 15], dtype=tf.float32)
            cs_prev = tf.random.uniform([3, 0], dtype=tf.float32)
            h_prev = tf.random.uniform([3, 0], dtype=tf.float32)
            w = tf.random.uniform([15, 0], dtype=tf.float32)
            wci = tf.random.uniform([0], dtype=tf.float32)
            wcf = tf.random.uniform([0], dtype=tf.float32)
            wco = tf.random.uniform([0], dtype=tf.float32)
            b = tf.random.uniform([0], dtype=tf.float32)
            res = tf.raw_ops.BlockLSTM(
                forget_bias=forget_bias,
                cell_clip=cell_clip,
                use_peephole=use_peephole,
                seq_len_max=seq_len_max,
                x=x,
                cs_prev=cs_prev,
                h_prev=h_prev,
                w=w,
                wci=wci,
                wcf=wcf,
                wco=wco,
                b=b,
            )
    except:
        pass
```

### Relevant log output

```shell
** On entry to SGEMM_EX  parameter number 8 had an illegal value
2022-10-20 10:48:03.831248: W tensorflow/core/framework/op_kernel.cc:1830] OP_REQUIRES failed at blas_gemm.cc:52 : INTERNAL: cublas error
2022-10-20 10:48:03.831307: F tensorflow/core/kernels/rnn/lstm_ops_gpu.cu.cc:277] Non-OK-status: GpuLaunchKernel( lstm_gates<T, false, gate_layout>, grid_dim_2d, block_dim_2d, 0, cu_stream, gates.data(), b.data(), cs_prev.data(), wci.data(), wcf.data(), wco.data(), o.data(), h.data(), ci.data(), cs.data(), co.data(), i.data(), f.data(), forget_bias, cell_clip, batch_size, cell_size) status: INTERNAL: invalid configuration argument
Aborted (core dumped)
```
</details>

**Comment by tiruk007** [comment]
@sushreebarsa
I was able to reproduce the issue on Colab using TF v2.10 in CPU and GPU. Please find the gist [CPU](https://colab.research.google.com/gist/tiruk007/4ad9adbd12f185a6e8e21872f30456e4/58176cpu.ipynb) and [GPU](https://colab.research.google.com/gist/tiruk007/cea5f6de11f9a26dd5e126ffffab0344/58176gpu.ipynb) for reference.
Thank you!

**Comment by sachinprasadhs** [comment]
tf.raw_ops are basically designed to be consumed by downstream library users, not end users.
Usually, these APIs are less strict in terms of validation etc., which is fine since mainly library writers are supposed to use these symbols.
If available, please prefer high level API for general use case scenarios.
Refer this [RFC](https://github.com/tensorflow/community/blob/master/rfcs/20181225-tf-raw-ops.md#design-proposal) for more details. Thanks!

**Comment by SuryanarayanaY** [comment]
Hi @shijy16 ,

The issue got resolved in tf-nightly(2.16.0-dev20231129). Attaching [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/a3b3cdb453fb6f39e034dab1a48f7ad3/58176_gpu_nightly-2-16-dev.ipynb#scrollTo=HuHAtA059tZS) for reference. Now the Op is raising the intended exception. Thanks!
